core/src/main/java/org/pentaho/di/core/vfs/KettleVFS.java

```diff
@@ -162,6 +175,19 @@ public static FileObject getFileObject( String vfsFilename, VariableSpace space,
     }
   }
 
+  protected static boolean checkForScheme( String[] initialSchemes, boolean relativeFilename, String vfsFilename,
+                                         VariableSpace space, FileSystemOptions fsOptions )
+    throws IOException {
+    for ( int i = 0; i < initialSchemes.length && relativeFilename; i++ ) {
+      if ( vfsFilename.startsWith( initialSchemes[ i ] + ":" ) ) {
+        relativeFilename = false;
+        // We have a VFS URL, load any options for the file system driver
+        fsOptions = buildFsOptions( space, fsOptions, vfsFilename, initialSchemes[ i ] );
```

Comment:
I had to look at this for a couple minutes to realize that buildFsOptions is mutating the fsOptions.  The .checkForScheme method doesn't return fsOptions or do anything else with it, so there isn't a need to assign.